cocos/physics/physx/joints/physx-joint.ts

```diff
@@ -40,7 +40,33 @@ export class PhysXJoint implements IBaseConstraint {
     }
 
     setConnectedBody (v: RigidBody | null): void {
-        // TODO
+        if (this._connectedBody === v) return;
+
+        // // unregister old
+        const oldBody2 = this._connectedBody;
+        if (oldBody2) {
+            const oldSB2 = (oldBody2.body as PhysXRigidBody).sharedBody;
+            oldSB2.removeJoint(this, 1);
+        }
+
+        const sb = (this._rigidBody.body as PhysXRigidBody).sharedBody;
+        sb.removeJoint(this, 0);
+        sb.addJoint(this, 0); // add to inner body if this joint is not added to sb
+        if (v) {
+            const sb2 = (v.body as PhysXRigidBody).sharedBody;
+            setJointActors(this._impl, sb.impl, sb2.impl);
+            sb2.addJoint(this, 1); // add to new sb2
+        } else {
+            setJointActors(this._impl, sb.impl, null);
+        }
+
+        if (oldBody2) {
+            oldBody2.wakeUp(); // wake it up, or the old body will be sleep for a while
```

Comment:
sleeping?

Reply:
Yes, it's possible, especially when the input rigid body has just been created.